### PR TITLE
Test/fluid strategy

### PR DESCRIPTION
**Pull request by bayou020**
This pull request includes several changes to configuration files and updates to dependencies. The most important changes include adding RPC endpoints, updating the coverage command, and modifying the fork setup in tests.

Configuration updates:
* [`FluidStrategy`](diffhunk://#diff-bac743fe3d899998e32393af53af8cc7d28c89c3d7fabfa48b01361cf8d42d9bR11-R12): 93.2% of lines and 100% of functions were covered

* [`foundry.toml`](diffhunk://#diff-bac743fe3d899998e32393af53af8cc7d28c89c3d7fabfa48b01361cf8d42d9bR11-R12): Added RPC endpoints for Arbitrum.

Dependency updates:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L29-R31): Updated `husky` to version 9.1.7 and `solhint` to version 5.0.3.

Coverage command improvements:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L15-R15): Enhanced the `forge-coverage` command to include branch coverage and function end line derivation.

Test setup improvements:

* [`test/utils/BaseTest.sol`](diffhunk://#diff-43679d9b8712a413d2c02a2492da3951821751b92f3cc0469a16dee8b76adffdL77-R77): Modified the `setArbitrumFork` function to use `createSelectFork` with a block number parameter for more control over the fork setup.

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -4,7 +4,10 @@ out = 'out'
 build_info = true
 build_info_path = 'out/build-info'
 solc_version = "0.8.19"
+evm_version = 'cancun'
```
Comment (YashP16) [inline foundry.toml]: @arcantheon 
please add a comment on why we had to add this config

**Review by YashP16 — COMMENTED** [review]
Test cases look good

**Review by YashP16 — APPROVED** [review]
coverage looks good

File: foundry.toml (hunk 1 above)
Comment (arcantheon) [reply foundry.toml]: Added be8da3709f7ae67582554137f55d693800b8641d

File: test/strategy/FluidStrategy.t.sol [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+        strategy.setPTokenAddress(ASSET, P_TOKEN);
+        vm.stopPrank();
+    }
+
+    function test_CheckRewardEarned() public {
+        InitializableAbstractStrategy.RewardData[] memory rewardData = strategy.checkRewardEarned();
+        assertEq(rewardData.length, 0);
+    }
+
+    // function test_CheckBalance() public {
```
Comment (arcantheon) [inline test/strategy/FluidStrategy.t.sol]: Can remove comments

File: test/strategy/FluidStrategy.t.sol [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+    }
+
+    function test_RevertWhen_CallerNotVault() public useActor(0) {
+        vm.expectRevert(abi.encodeWithSelector(CallerNotVault.selector, actors[0]));
+        strategy.withdraw(VAULT, ASSET, 1);
+    }
+
+    function test_revertWhen_limitIsReached() public useKnownActor(VAULT) {
+        _depositHugeAmount();
+        // emit Withdrawal(ASSET, depositAmount * 10e6);
```
Comment (arcantheon) [inline test/strategy/FluidStrategy.t.sol]: Can remove comment

File: test/strategy/FluidStrategy.t.sol [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+
+    function testFuzz_Deposit(uint256 _depositAmount) public useKnownActor(VAULT) {
+        depositAmount = bound(_depositAmount, 1 * 10 ** ERC20(ASSET).decimals(), 1e6 * 10 ** ERC20(ASSET).decimals());
+        uint256 initial_bal = strategy.checkBalance(ASSET);
+        uint256 initialLPBalance = strategy.checkLPTokenBalance(ASSET);
+        emit log_named_uint("initial LP balance", initialLPBalance);
+        assert(initialLPBalance == 0);
+        deal(ASSET, VAULT, depositAmount);
+        IERC20(ASSET).approve(address(strategy), depositAmount);
+        strategy.deposit(ASSET, depositAmount);
```
Comment (arcantheon) [inline test/strategy/FluidStrategy.t.sol:295]: Expect emit is missing here, please add here and anywhere else it is missing.

File: test/strategy/FluidStrategy.t.sol [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+
+        (uint256 allocatedAmt) = strategy.allocatedAmount(ASSET);
+
+        assertEq(allocatedAmt, 0);
+        assertEq(strategy.assetToPToken(ASSET), address(0));
+        assertFalse(strategy.supportsCollateral(ASSET));
+    }
+}
+
+contract DepositTest is FluidStrategyTest {
```
Comment (arcantheon) [inline test/strategy/FluidStrategy.t.sol:249]: Add one test case where allocate is called from the vault and vault is calling deposit on strategy. Real world use case.

Comment (arcantheon) [reply test/strategy/FluidStrategy.t.sol:249]: Same for withdraw

File: test/strategy/FluidStrategy.t.sol [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+    FluidStrategy internal strategy;
+    FluidStrategy internal impl;
+    UpgradeUtil internal upgradeUtil;
+    uint256 internal depositAmount;
+    uint256 internal interestAmount;
+    address internal proxyAddress;
+    address internal yieldReceiver;
+    address internal ASSET;
+    address internal P_TOKEN;
+    uint16 internal constant depositSlippage = 200;
```
Comment (arcantheon) [inline test/strategy/FluidStrategy.t.sol]: Update the slippage to 50 (0.5%) which is acceptable slippage.

PS: I have checked it the test cases work fine, means the slippage under 0.5% works. just need to update it.

Comment (arcantheon) [reply test/strategy/FluidStrategy.t.sol]: Please update `withdrawSlippage` as well.

Comment (bayou020) [reply test/strategy/FluidStrategy.t.sol]: Fixed

File: foundry.toml [hunk 7]
```diff
@@ -4,7 +4,10 @@ out = 'out'
...
 solc_version = "0.8.19"
+evm_version = 'cancun'   # Some of the test cases failing on AaveStrategy's test cases are working fine by explicitly mentioning this.
 
 libs = ['node_modules', 'lib']
 test = 'test'
-cache_path  = 'cache_forge'
\ No newline at end of file
+cache_path  = 'cache_forge'
+[rpc_endpoints]
+arbitrum = 'https://arb1.arbitrum.io/rpc'
```
Comment (parv3213) [inline foundry.toml:13]: What is the significance of this, where is this used?

Comment (bayou020) [reply foundry.toml:13]: @parv3213 i was referring to [this](https://book.getfoundry.sh/cheatcodes/rpc), If that has no effect, I can remove it

Comment (parv3213) [reply foundry.toml:13]: If we are not using it, we can remove it.

Comment (bayou020) [reply foundry.toml:13]: Fixed @parv3213 please resolve it

**Pull request merged**